common/unit-test/config_unittest.cpp

```diff
@@ -0,0 +1,196 @@
+ * from file.
+ */
+class StringConfigLoader : public ConfigLoaderAbstractInterface
+{
+
+LOG_DECLARE_CLASS_CONTEXT("CFGL", "SoftwareContainer general config loader");
+
+public:
+    // Constructor just needs to init parent with the config source string
+    StringConfigLoader(const std::string &source) : ConfigLoaderAbstractInterface(source) {}
+
+    std::unique_ptr<Glib::KeyFile> loadConfig() override
+    {
+        std::unique_ptr<Glib::KeyFile> configData = std::unique_ptr<Glib::KeyFile>(new Glib::KeyFile);
+        try {
+            configData->load_from_data(Glib::ustring(this->m_source), Glib::KEY_FILE_NONE);
+        } catch (Glib::KeyFileError &error) {
+            log_error() << "Could not load SoftwareContainer config: \"" << error.what() << "\"";
+            throw error;
+        }
+
+        return configData;
+    }
+};
+
+
+/*
+ * StringConfigLoaderTest suite
+ *
+ * NOTE: Tests the test stub...
+ *
+ * This suite does some basic tests for the StringConfigLoader used for testing
+ * the Config class. The purpose is to support troubleshooting between the test
+ * setup and the Config class which is tested in the other suite.
+ */
+class StringConfigLoaderTest : public ::testing::Test
+{
+};
+
+/*
+ * Test that a well formed config will not cause an exception to be thrown.
+ */
+TEST_F(StringConfigLoaderTest, SuccessfulLoadDoesNotThrow) {
+    // Well formed config
+    const std::string configString = "[SoftwareContainer]\n"
+                                     "sc.foo=bar\n";
+
+    StringConfigLoader loader(configString);
+    ASSERT_NO_THROW(loader.loadConfig());
+}
+
+/*
+ * Test that a broken config string will cause an exception to be thrown.
+ */
+TEST_F(StringConfigLoaderTest, UnsuccessfulLoadThrows) {
+    // This config is broken, it's missing a bracket in the group
+    const std::string configString = "[SoftwareContainer\n"
+                                     "sc.foo=bar\n";
+
+    StringConfigLoader loader(configString);
+    ASSERT_THROW(loader.loadConfig(), Glib::KeyFileError);
+}
+
+/*
+ * Test that a broken config string will cause an exception to be thrown, and that
+ * the error code thrown is the correct one.
+ */
+TEST_F(StringConfigLoaderTest, UnsuccessfulLoadThrowsExpectedErrorCode) {
+    // This config is broken, it's missing a bracket in the group
+    const std::string configString = "[SoftwareContainer\n"
+                                     "sc.foo=bar\n";
+
+    StringConfigLoader loader(configString);
+
+    try {
+        std::unique_ptr<Glib::KeyFile> config = loader.loadConfig();
+    } catch (Glib::KeyFileError &error) {
+        ASSERT_TRUE(error.code() == Glib::KeyFileError::PARSE);
+    }
+}
+
+
+/*
+ * ConfigTest suite
+ *
+ * This suite tests the Config class. It uses a stubbed loader, the StringConfigLoader, to
+ * provide the config source, and does asserts on the way Config parses the config provided.
+ */
+class ConfigTest : public ::testing::Test
+{
+};
```

Comment:
This fixture class isn't very useful. Consider moving your instantiation of the loader and config objects into the fixture class.